pkg/service/service_binding_builder.go

```diff
@@ -0,0 +1,207 @@
+// we can go back to using the builder.DefaultBuilder instead of the OdoDefaultBuilder in getPipeline.
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/klog"
+
+	"github.com/redhat-developer/service-binding-operator/apis"
+	"github.com/redhat-developer/service-binding-operator/pkg/binding"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/builder"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/collect"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/mapping"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/naming"
+	"github.com/redhat-developer/service-binding-operator/pkg/reconcile/pipeline/handler/project"
+	"github.com/redhat-developer/service-binding-operator/pkg/util"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var defaultFlow = []pipeline.Handler{
+	pipeline.HandlerFunc(project.Unbind),
+	pipeline.HandlerFunc(collect.PreFlight),
+	pipeline.HandlerFunc(ProvisionedService),
+	pipeline.HandlerFunc(collect.DirectSecretReference),
+	pipeline.HandlerFunc(BindingDefinitions),
+	pipeline.HandlerFunc(collect.BindingItems),
+	pipeline.HandlerFunc(collect.OwnedResources),
+	pipeline.HandlerFunc(mapping.Handle),
+	pipeline.HandlerFunc(naming.Handle),
+	pipeline.HandlerFunc(project.PreFlightCheck()),
+	pipeline.HandlerFunc(project.InjectSecretRef),
+	pipeline.HandlerFunc(project.BindingsAsEnv),
+	pipeline.HandlerFunc(project.BindingsAsFiles),
+	pipeline.HandlerFunc(project.PostFlightCheck),
+}
+
+var OdoDefaultBuilder = builder.Builder().WithHandlers(defaultFlow...)
+
+func ProvisionedService(ctx pipeline.Context) {
+	services, _ := ctx.Services()
+
+	for _, service := range services {
+		res := service.Resource()
+		secretName, found, err := unstructured.NestedString(res.Object, "status", "binding", "name")
+		if err != nil {
+			requestRetry(ctx, collect.ErrorReadingBindingReason, err)
+			return
+		}
+		if found {
+			if secretName != "" {
+				secret, err := ctx.ReadSecret(res.GetNamespace(), secretName)
+				if err != nil {
+					requestRetry(ctx, collect.ErrorReadingSecret, err)
+					return
+				}
+				ctx.AddBindings(&pipeline.SecretBackedBindings{Service: service, Secret: secret})
+			}
+		} else {
+			crd, err := service.CustomResourceDefinition()
+			if err != nil {
+				// If the user does not have permission to access CRD, ignore
+				if !kerrors.IsForbidden(err) {
+					requestRetry(ctx, collect.ErrorReadingCRD, err)
+					return
+				} else {
+					klog.V(4).Infof("Skipping the check for CRD, user does not have access")
```

Comment:
Can we make this user normal facing message? Ideally a warning?
odo user should know that something like this was skipped without running odo with increased log level.